clients/src/main/java/org/apache/kafka/common/protocol/types/ArrayOf.java

```diff
@@ -72,6 +94,9 @@ public String toString() {
     @Override
     public Object[] validate(Object item) {
         try {
+            if (isNullable() && item == null)
+                return null;
```

Comment:
I can do that. It should probably be done for all Types that don't accept null. I opened [KAFKA-3603](https://issues.apache.org/jira/browse/KAFKA-3603) to track that.